### ISSUE TITLE
Crash on import of GLB

### ISSUE THREAD

**Issue by Wavesonics** [godotengine/godot#44017]
**Godot version:**
3.2.3

**OS/device including version:**
Win64, Linux64

**Issue description:**
Importing a particular GLB file crashes the editor.

The glb in question here validates here: https://github.khronos.org/glTF-Validator/

So I don't think it's a problem with the file.

The crash is in: `xatlas.cpp` **line 814**
```cpp
static Vector3 normalize(const Vector3 &v, float epsilon)
{
    float l = length(v);
    XA_DEBUG_ASSERT(!isZero(l, epsilon));
```

In this case `l` is a very small number: `2.92496161e-08`
v: `x: 0.0 y: 2.92496161e-08 z: 0.0`

Here's the stack trace:
```
Unhandled exception at 0x034011E2 in godot.windows.tools.32.exe: Fatal program exit requested. 	godot.windows.tools.32.exe!abort() Line 77	C++
 	godot.windows.tools.32.exe!common_assert_to_stderr_direct(const wchar_t * const expression, const wchar_t * const file_name, const unsigned int line_number) Line 160	C++
 	godot.windows.tools.32.exe!common_assert_to_stderr<wchar_t>(const wchar_t * const expression, const wchar_t * const file_name, const unsigned int line_number) Line 179	C++
 	[Inline Frame] godot.windows.tools.32.exe!common_assert(const wchar_t * const return_address, const wchar_t * const) Line 420	C++
 	godot.windows.tools.32.exe!_wassert(const wchar_t * expression, const wchar_t * file_name, unsigned int line_number) Line 443	C++
>	godot.windows.tools.32.exe!xatlas::internal::normalize(const xatlas::internal::Vector3 & v, float epsilon) Line 814	C++
 	godot.windows.tools.32.exe!xatlas::internal::param::projectTriangle(xatlas::internal::Vector3 p0, xatlas::internal::Vector3 p1, xatlas::internal::Vector3 p2, xatlas::internal::Vector2 * z0, xatlas::internal::Vector2 * z1, xatlas::internal::Vector2 * z2, float epsilon) Line 6618	C++
 	godot.windows.tools.32.exe!xatlas::internal::param::computeLeastSquaresConformalMap(xatlas::internal::Mesh * mesh) Line 6656	C++
 	godot.windows.tools.32.exe!xatlas::internal::param::Chart::parameterize(const xatlas::ParameterizeOptions & options, xatlas::internal::UniformGrid2 & boundaryGrid) Line 7512	C++
 	godot.windows.tools.32.exe!xatlas::internal::param::runCreateAndParameterizeChartTask(void * userData) Line 7632	C++
 	godot.windows.tools.32.exe!xatlas::internal::TaskScheduler::workerThread(xatlas::internal::TaskScheduler * scheduler, xatlas::internal::TaskScheduler::Worker * worker, unsigned int threadIndex) Line 3892	C++

```

It looks like the problem is that the value is smaller than the provided epsilon. So it's treated as zero.

**Steps to reproduce:**
Attempt to open the sample project:
[ImportCrash.zip](https://github.com/godotengine/godot/files/5625494/ImportCrash.zip)

**Minimal reproduction project:**
ImportCrash.zip

**Comment by Wavesonics** [comment]
I tested this in master and it still crashes.

However when I apply my fix from #44020 to master, it does fix the crash, but it is all black, and has new errors in the terminal:
```
ERROR: 1 shaders of type CanvasSdfShaderRD were never freed
   at: ShaderRD::~ShaderRD (servers\rendering\rasterizer_rd\shader_rd.cpp:470)
ERROR: Invalid Object id '25367151136', can't call method.
   at: (D:\Code\External\godot\core/object/callable_method_pointer.h:95)

ERROR: Condition "p_size < 0" is true. Returning: ERR_INVALID_PARAMETER
   at: CowData<struct Vector2>::resize (D:\Code\External\godot\core/templates/cowdata.h:252)
ERROR: Cannot open file 'res://.import/cop_car.glb-d1da082c95816014f295a29160723f8f.scn'.
   at: (core\io\resource_format_binary.cpp:1030)
Adding mesh..
Generate..
Final texture size: 134.000000,134.000000 - max 133.500000,133.500000
ERROR: Another resource is loaded from path 'res://cop_car/police_car.mesh' (possible cyclic resource inclusion).
   at: (core\io\resource.cpp:79)
ERROR: Another resource is loaded from path 'res://cop_car/police_car.mesh' (possible cyclic resource inclusion).
   at: (core\io\resource.cpp:79)
ERROR: Another resource is loaded from path 'res://cop_car/police_car.mesh' (possible cyclic resource inclusion).
   at: (core\io\resource.cpp:79)
ERROR: Another resource is loaded from path 'res://cop_car/police_car.mesh' (possible cyclic resource inclusion).
   at: (core\io\resource.cpp:79)
ERROR: Another resource is loaded from path 'res://cop_car/police_car.mesh' (possible cyclic resource inclusion).
   at: (core\io\resource.cpp:79)
ERROR: Another resource is loaded from path 'res://cop_car/police_car.mesh' (possible cyclic resource inclusion).
   at: (core\io\resource.cpp:79)
ERROR: Another resource is loaded from path 'res://cop_car/police_car.mesh' (possible cyclic resource inclusion).
   at: (core\io\resource.cpp:79)
ERROR: Another resource is loaded from path 'res://cop_car/police_car.mesh' (possible cyclic resource inclusion).
   at: (core\io\resource.cpp:79)
ERROR: Another resource is loaded from path 'res://cop_car/police_car.mesh' (possible cyclic resource inclusion).
   at: (core\io\resource.cpp:79)
ERROR: Another resource is loaded from path 'res://cop_car/police_car.mesh' (possible cyclic resource inclusion).
   at: (core\io\resource.cpp:79)
ERROR: Texture (binding: 0, index 0) is not a valid texture.
   at: (drivers\vulkan\rendering_device_vulkan.cpp:4680)
ERROR: Condition "!uniform_set" is true.
   at: RenderingDeviceVulkan::draw_list_bind_uniform_set (drivers\vulkan\rendering_device_vulkan.cpp:6143)
ERROR: Uniforms were never supplied for set (3) at the time of drawing, which are required by the pipeline
   at: (drivers\vulkan\rendering_device_vulkan.cpp:6285)
ERROR: Texture (binding: 0, index 0) is not a valid texture.
   at: (drivers\vulkan\rendering_device_vulkan.cpp:4680)
ERROR: Condition "!uniform_set" is true.
   at: RenderingDeviceVulkan::draw_list_bind_uniform_set (drivers\vulkan\rendering_device_vulkan.cpp:6143)
ERROR: Uniforms were never supplied for set (3) at the time of drawing, which are required by the pipeline
   at: (drivers\vulkan\rendering_device_vulkan.cpp:6285)
ERROR: Texture (binding: 0, index 0) is not a valid texture.
   at: (drivers\vulkan\rendering_device_vulkan.cpp:4680)
ERROR: Condition "!uniform_set" is true.
   at: RenderingDeviceVulkan::draw_list_bind_uniform_set (drivers\vulkan\rendering_device_vulkan.cpp:6143)
ERROR: Uniforms were never supplied for set (3) at the time of drawing, which are required by the pipeline
   at: (drivers\vulkan\rendering_device_vulkan.cpp:6285)
ERROR: Texture (binding: 0, index 0) is not a valid texture.
   at: (drivers\vulkan\rendering_device_vulkan.cpp:4680)
ERROR: Condition "!uniform_set" is true.
   at: RenderingDeviceVulkan::draw_list_bind_uniform_set (drivers\vulkan\rendering_device_vulkan.cpp:6143)
ERROR: Uniforms were never supplied for set (3) at the time of drawing, which are required by the pipeline
   at: (drivers\vulkan\rendering_device_vulkan.cpp:6285)
ERROR: Texture (binding: 0, index 0) is not a valid texture.
   at: (drivers\vulkan\rendering_device_vulkan.cpp:4680)
ERROR: Condition "!uniform_set" is true.
   at: RenderingDeviceVulkan::draw_list_bind_uniform_set (drivers\vulkan\rendering_device_vulkan.cpp:6143)
ERROR: Uniforms were never supplied for set (3) at the time of drawing, which are required by the pipeline
   at: (drivers\vulkan\rendering_device_vulkan.cpp:6285)
ERROR: Texture (binding: 0, index 0) is not a valid texture.
   at: (drivers\vulkan\rendering_device_vulkan.cpp:4680)
ERROR: Condition "!uniform_set" is true.
   at: RenderingDeviceVulkan::draw_list_bind_uniform_set (drivers\vulkan\rendering_device_vulkan.cpp:6143)
ERROR: Uniforms were never supplied for set (3) at the time of drawing, which are required by the pipeline
   at: (drivers\vulkan\rendering_device_vulkan.cpp:6285)
ERROR: Texture (binding: 0, index 0) is not a valid texture.
   at: (drivers\vulkan\rendering_device_vulkan.cpp:4680)
ERROR: Condition "!uniform_set" is true.
   at: RenderingDeviceVulkan::draw_list_bind_uniform_set (drivers\vulkan\rendering_device_vulkan.cpp:6143)
ERROR: Uniforms were never supplied for set (3) at the time of drawing, which are required by the pipeline
   at: (drivers\vulkan\rendering_device_vulkan.cpp:6285)
ERROR: Texture (binding: 0, index 0) is not a valid texture.
   at: (drivers\vulkan\rendering_device_vulkan.cpp:4680)
ERROR: Condition "!uniform_set" is true.
   at: RenderingDeviceVulkan::draw_list_bind_uniform_set (drivers\vulkan\rendering_device_vulkan.cpp:6143)
ERROR: Uniforms were never supplied for set (3) at the time of drawing, which are required by the pipeline
   at: (drivers\vulkan\rendering_device_vulkan.cpp:6285)
ERROR: Texture (binding: 0, index 0) is not a valid texture.
   at: (drivers\vulkan\rendering_device_vulkan.cpp:4680)
ERROR: Condition "!uniform_set" is true.
   at: RenderingDeviceVulkan::draw_list_bind_uniform_set (drivers\vulkan\rendering_device_vulkan.cpp:6143)
ERROR: Uniforms were never supplied for set (3) at the time of drawing, which are required by the pipeline
   at: (drivers\vulkan\rendering_device_vulkan.cpp:6285)
ERROR: Texture (binding: 0, index 0) is not a valid texture.
   at: (drivers\vulkan\rendering_device_vulkan.cpp:4680)
ERROR: Condition "!uniform_set" is true.
   at: RenderingDeviceVulkan::draw_list_bind_uniform_set (drivers\vulkan\rendering_device_vulkan.cpp:6143)
ERROR: Uniforms were never supplied for set (3) at the time of drawing, which are required by the pipeline
   at: (drivers\vulkan\rendering_device_vulkan.cpp:6285)
ERROR: Texture (binding: 0, index 0) is not a valid texture.
   at: (drivers\vulkan\rendering_device_vulkan.cpp:4680)
ERROR: Condition "!uniform_set" is true.
   at: RenderingDeviceVulkan::draw_list_bind_uniform_set (drivers\vulkan\rendering_device_vulkan.cpp:6143)
ERROR: Uniforms were never supplied for set (3) at the time of drawing, which are required by the pipeline
   at: (drivers\vulkan\rendering_device_vulkan.cpp:6285)
ERROR: Texture (binding: 0, index 0) is not a valid texture.
   at: (drivers\vulkan\rendering_device_vulkan.cpp:4680)
ERROR: Condition "!uniform_set" is true.
   at: RenderingDeviceVulkan::draw_list_bind_uniform_set (drivers\vulkan\rendering_device_vulkan.cpp:6143)
ERROR: Uniforms were never supplied for set (3) at the time of drawing, which are required by the pipeline
   at: (drivers\vulkan\rendering_device_vulkan.cpp:6285)
ERROR: Texture (binding: 0, index 0) is not a valid texture.
   at: (drivers\vulkan\rendering_device_vulkan.cpp:4680)
ERROR: Condition "!uniform_set" is true.
   at: RenderingDeviceVulkan::draw_list_bind_uniform_set (drivers\vulkan\rendering_device_vulkan.cpp:6143)
ERROR: Uniforms were never supplied for set (3) at the time of drawing, which are required by the pipeline
   at: (drivers\vulkan\rendering_device_vulkan.cpp:6285)
```

**Comment by Wavesonics** [comment]
The fix here was to pull in the latest version of xatlas.
